packages/engine/src/renderer/components/HighlightComponent.ts

```diff
@@ -39,29 +40,52 @@ export const HighlightComponent = defineComponent({
 
   reactor: function () {
     const entity = useEntityContext()
+    const xrSession = useHookstate(getMutableState(XRState)).session.value
 
     const postProcessingSettingsState = useHookstate(getMutableState(PostProcessingSettingsState))
     const usePostProcessing = useHookstate(getMutableState(RendererState).usePostProcessing)
     const group = useComponent(entity, GroupComponent)
 
     useEffect(() => {
       const objs = [...group.value]
-      for (const object of objs) object.traverse(addToSelection)
+      for (const object of objs)
+        object.traverse((obj) => {
+          obj.type === 'Mesh' && addToSelection(obj as Mesh, xrSession)
+        })
       return () => {
-        for (const object of objs) object.traverse(removeFromSelection)
+        for (const object of objs)
+          object.traverse((obj) => {
+            obj.type === 'Mesh' && removeFromSelection(obj as Mesh, xrSession)
+          })
       }
     }, [group, postProcessingSettingsState.effects, postProcessingSettingsState.enabled, usePostProcessing])
 
     return null
   }
 })
 
-const addToSelection = (obj: Object3D) => {
+const animateScale = (obj: Mesh, v: Vector3, xrSession) => {
+  let start = 0.0
+  function animate(timeStamp) {
+    if (start === 0.0) start = timeStamp
+    const elapsed = timeStamp - start
+    obj.scale.lerp(v, MathUtils.damp(0, 1, 0.01, elapsed))
+
+    if (obj.scale.x !== v.x) {
+      !xrSession ? requestAnimationFrame(animate) : xrSession?.requestAnimationFrame(animate)
```

Comment:
We don't want to be creating new animation frame requests, this should be handled in a system.

Reply:
for the sake of time. I've reverted the changes in case this is needed. Will still be working on the animation for UX reasons.

Reply:
Alright resolved with the TweenComponent setup. Was stuck for too long tweening between 1-0 for lerp 😢